src/scales/scale.time.js

```diff
@@ -361,27 +361,54 @@ function generate(min, max, capacity, options) {
  * Returns the right and left offsets from edges in the form of {left, right}.
  * Offsets are added when the `offset` option is true.
  */
-function computeOffsets(table, ticks, min, max, options) {
+function computeOffsets(table, ticks, data, min, max, options) {
+	var pos = [];
+	var minInterval = 1;
+	var timeOpts = options.time;
+	var barThickness = options.barThickness;
 	var left = 0;
 	var right = 0;
-	var upper, lower;
-
-	if (options.offset && ticks.length) {
-		if (!options.time.min) {
-			upper = ticks.length > 1 ? ticks[1] : max;
-			lower = ticks[0];
-			left = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
+	var i, ilen, curr, prev, length, width;
+
+	if (options.offset) {
+		data.forEach(function(timestamp) {
+			if (timestamp >= min && timestamp <= max) {
+				pos.push(interpolate(table, 'time', timestamp, 'pos'));
+			}
+		});
+
+		if (!barThickness) {
+			[data, ticks].forEach(function(timestamps) {
```

Comment:
can you add a comment here:

    // Calculate minInterval

Reply:
Yes, I will add this.